src/main/java/com/labsynch/labseer/chemclasses/bbchem/ChemStructureServiceBBChemImpl.java

```diff
@@ -1233,4 +1232,134 @@ public StandardizationSettingsConfigCheckResponseDTO checkStandardizerSettings(S
 		}
 	}
 
+
+	@Transactional
+	public void populateDuplicateChangedStructures(int batchSize) {
+		// Update existing, new and changed structures
+		updateExistingDuplicates();
+		updateNewDuplicates();
+		updateChangedStructure();
+	}
+	
+	@Transactional
+	public static void updateExistingDuplicates() {
+		// Existing Duplicates: List of parent corp names that were already the same structure as the standardized structure taking into account tautomers, stereo comments and stereo categories.
+		EntityManager em = StandardizationDryRunCompound.entityManager();
+	
+		String sql = "WITH old_duplicates AS ( " +
+					 "    SELECT " +
+					 "        p1.id AS parent_id, " +
+					 "        COUNT(DISTINCT p2.corp_name) AS duplicate_count, " +
+					 "        STRING_AGG(DISTINCT p2.corp_name, ';') AS duplicate_corp_names " +
+					 "    FROM " +
+					 "        parent p1 " +
+					 "    JOIN " +
+					 "        bbchem_parent_structure bps1 ON p1.cd_id = bps1.id " +
+					 "    JOIN " +
+					 "        bbchem_parent_structure bps2 ON bps1.reg = bps2.reg " +
+					 "    JOIN " +
+					 "        parent p2 ON bps2.id = p2.cd_id " +
+					 "    WHERE " +
+					 "        p1.corp_name != p2.corp_name " +
+					 "        AND p1.stereo_category = p2.stereo_category " +
+					 "        AND ( " +
+					 "            (p1.stereo_comment IS NULL AND p2.stereo_comment IS NULL) OR " +
+					 "            (p1.stereo_comment IS NOT NULL AND p2.stereo_comment IS NOT NULL AND " +
+					 "             LOWER(p1.stereo_comment) = LOWER(p2.stereo_comment)) " +
+					 "        ) " +
+					 "    GROUP BY " +
+					 "        p1.id " +
+					 ") " +
+					 "UPDATE standardization_dry_run_compound sdr " +
+					 "SET " +
+					 "    existing_duplicate_count = old_duplicates.duplicate_count, " +
+					 "    existing_duplicates = old_duplicates.duplicate_corp_names " +
+					 "FROM " +
+					 "    old_duplicates " +
+					 "WHERE " +
+					 "    sdr.parent_id = old_duplicates.parent_id " +
+					 "    AND sdr.existing_duplicate_count IS NULL;";
+		em.createNativeQuery(sql).executeUpdate();
+	}
+	
+	@Transactional
+	public static void updateNewDuplicates() {
+		// New Duplicates: List of parent corp names that are now the same structure as the standardized structure taking into account tautomers, stereo comments and stereo categories.
+		EntityManager em = StandardizationDryRunCompound.entityManager();
+	
+		String newDuplicatesSql = "WITH new_duplicates AS ( " +
+								  "    SELECT " +
+								  "        p1.id AS parent_id, " +
+								  "        COUNT(DISTINCT p2.corp_name) AS duplicate_count, " +
+								  "        STRING_AGG(DISTINCT p2.corp_name, ';') AS duplicate_corp_names " +
+								  "    FROM " +
+								  "        parent p1 " +
+								  "    JOIN " +
+								  "        standardization_dry_run_compound sdr1 ON p1.id = sdr1.parent_id " +
+								  "    JOIN " +
+								  "        bbchem_parent_structure bps1 ON p1.cd_id = bps1.id " +
```

Comment:
Is this join to `bbchem_parent_structure` needed? It doesn't look like we use its data anywhere else in the query

Reply:
Thanks, I missed that. Fixed.  Had one large query originally and decided to split it and missed the extra join.